### ISSUE TITLE
SIGSEGV on launch develop version

### ISSUE THREAD

**Issue by Fusion86** [DegateCommunity/Degate#21]
**Describe the bug**
The latest develop release throws a SIGSEGV when starting Degate.

**To Reproduce**
1. Build degate with Qt 6.5.2 GCC 64bit
2. Launch the newly built executable in build/out/bin
3. Critical error message.

```
Build ABI: 'x86_64-little_endian-lp64'
Kernel type: 'linux'
kernel version: '5.4.0-159-generic'
Current CPU architecture: 'x86_64'
Build CPU architecture: 'x86_64'
Product type: 'ubuntu'
Pretty product name: 'Ubuntu 20.04.6 LTS'
Product version: '20.04'
Degate version: '2.1.0-alpha.1'
Degate version type: 'nightly'
Degate release date: 'unreleased'
An unexpected error occurred. Error message:

'A SIGSEGV occurred (11).'
```

**Possible fix**
In `MainWindow.cc` the call `QGuiApplication::screenAt(this->pos())` returns a nullptr on my machine. Not sure why, but it does. If I comment out that line (or add nullptr checks) the application starts just fine.
Possible fix -> https://github.com/Fusion86/Degate/commit/0017bcb008fcd9ff0ea70da503308a294d73aacb

Just reporting this here for completeness sake. I fixed my own issue, so feel free to close this if this doesn't belong here.

**Comment by DorianBDev** [comment]
Hello, thanks for the bug report (and identifying the source). This seems to be related to #19 (Qt6 migration), something may have changed in Qt6. I will merge a fix in the next few minutes

**Comment by DorianBDev** [comment]
@Fusion86 This should be fixed now with 3dc2861, can you confirm? Also thank you for other contributions related to deprecation warnings

**Comment by Fusion86** [comment]
Yes, this fixes the issue for me. Thanks!
